utils/controller.js

```diff
@@ -0,0 +1,296 @@
+    {
+        return res.json({msg :"kitne kareedega?"});
+    }
+    })
+    
+    Company.findById(req.body.companyId,(err,companys)=>{
+        if(err) res.json(err);
+        else{
+            var historytemp  = {
+                sharePrice : companys.sharePrice,
+                availableQuantity : companys.availableQuantity+req.body.NoOfShares
+            }
+            var activitytemp ={
+                company : companys._id,
+                action : "shorted" ,
+    	        quantity : req.body.NoOfShares,
+    	        price : companys.sharePrice
+            }
+            companys.history.push(historytemp);
+            companys.availableQuantity += req.body.NoOfShares;
+            user.activity.push(activitytemp);
+            user.accountBalance += req.body.NoOfShares*companys.sharePrice;
+            if(user.stockShorted.id(companys._id))
+            {
+                
+                user.stockShorted.id(companys._id).quantity += req.body.NoOfShares;
+            }
+            else{
+                user.stockShorted.push({_id : companys._id,quantity : req.body.NoOfShares});
+            }
+        }
+        companys.save();
+        user.save();
+        return res.json({success :"true",Company : companys,Customer : user});
+    })
+
+},
+coverstock(req,res){
+
+    let user ;
+    User.findById(req.body.id,(err,data)=>{
+        if(err) throw err;
+        user = data;})
+    
+    Company.findById(req.body.companyId,(err,companys)=>{
+        if(err) res.json(err);
+        if(!user.stockShorted.id(companys._id))
+        {
+            return res.json({msg :"shorted stock nhi he"});
+        }
+        if(user.stockShorted.id(companys._id).quantity<req.body.NoOfShares){
+           return res.json({msg :"itne shorted stock nhi he"});
+        }
+        else{
+            var historytemp  = {
+                sharePrice : companys.sharePrice,
+                availableQuantity : companys.availableQuantity-req.body.NoOfShares
+            }
+            var activitytemp ={
+                company : companys._id,
+                action : "covered" ,
+    	        quantity : req.body.NoOfShares,
+    	        price : companys.sharePrice
+            }
+            companys.history.push(historytemp);
+            user.accountBalance -= companys.sharePrice*req.body.NoOfShares;
+            companys.availableQuantity -= req.body.NoOfShares;
+            user.activity.push(activitytemp);
+            if(user.stockShorted.id(companys._id))
+            {
+                user.stockShorted.id(companys._id).quantity -= req.body.NoOfShares;
+            }
+            
+        }
+        companys.save();
+        user.save();
+        return res.json({success :"true",Company : companys,Customer : user});
+    })
+
+},
+takeloan(req,res)
+{
+    let user ;
+    User.findById(req.body.id,(err,data)=>{
+        if(err) throw err;
+        user = data;
+        if(user.loan[0].amount+req.body.loan>parameter.maxLoan)
```

Comment:
User is not able to take loan twice even when this sum is less that maxLoan. Figure out why.